src/KafkaNET.Library/ZooKeeperIntegration/Listeners/ZKRebalancerListener.cs

```diff
@@ -540,32 +525,6 @@ private void AddPartitionTopicInfo(ZKGroupTopicDirs topicDirs, string partition,
             Logger.InfoFormat("{0} selected new offset {1}", partTopicInfo, offset);
         }
 
-        private long EarliestOrLatestOffset(string topic, int brokerId, int partitionId, long earliestIoLatest)
-        {
-            Consumer consumer = null;
-            long producedOffset = -1;
-            try
-            {
-                var cluster = new Cluster(this.zkClient);
-                var broker = cluster.GetBroker(brokerId);
-                if (broker == null)
-                {
-                    throw new IllegalStateException(string.Format("Broker {0} is unavailable. Cannot issue GetOffsetsBefore request", brokerId));
-                }
-                consumer = new Consumer(this.config, broker.Host, broker.Port);
-                var requestInfos = new Dictionary<string, List<PartitionOffsetRequestInfo>>();
-                requestInfos[topic] = new List<PartitionOffsetRequestInfo>() { new PartitionOffsetRequestInfo(partitionId, earliestIoLatest, 1) };
-                var offsets = consumer.GetOffsetsBefore(new OffsetRequest(requestInfos));
```

Comment:
If kafka connection here threw an exception (say temporary network glitch), then we would return -1, which would force auto-reset of offsets (which could lead to log loss if it jumps to latest offset)